oauth2-useragent-javafx/src/main/java/com/microsoft/alm/oauth2/useragent/InterceptingBrowser.java

```diff
@@ -113,6 +115,10 @@ static boolean matchesRedirection(final String expectedRedirectUriString, final
             }
         }
 
+        if (actualUri.getScheme().equals("urn") && !actualUriString.startsWith(expectedRedirectUriString)) {

```

Comment:
The URL created from string `urn:ietf:wg:oauth:2.0:oob?code=abc&stat=123` only has scheme, query param and scheme specific parts; It doesn't have host, port, nor path components.    

So even if we use the other constructor to create a URL (instead of URI), we still have to do a string comparison.  

That said I am not opposed to have one way to create redirect URLs though.